shadps4: 0.13.0 -> 0.15.0, shadps4-qtlauncher: init at 224

ryand56 opened the pull request:
https://github.com/shadps4-emu/shadPS4/releases/tag/v.0.15.0
Diff: https://github.com/shadps4-emu/shadPS4/compare/v.0.13.0...v.0.15.0

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ryand56 committed:
shadps4-qt: init at 224

ryand56 committed:
shadps4-qtlauncher: init at 224

Sirius902 commented:
@ryand56 This seems to work for me but I'd like to add that it seems opening the `.desktop` doesn't work but adding this to the `installPhase` or `fixupPhase` should fix it, feel free to throw it in this PR.
```sh
substituteInPlace $out/share/applications/net.shadps4.shadps4-qtlauncher.desktop \
  --replace-fail 'Exec=shadPS4QtLauncher' "Exec=''${!outputBin}/bin/shadps4-qt"
```

Also it seems audio doesn't work for me, not sure if this is an issue with upstream shadPS4 or not though.

ryand56 commented:
> [ryand56](https://github.com/ryand56)
> 
> Hi Could you please tell if you've got to launch games on Wayland? I've built and ran with your changes and got "Failed to initialize SDL video subsystem: wayland not available"

Are you using X11 or Wayland? I tested with the OpenOrbis demo and it works

ryand56 commented:
> > > Are you using X11 or Wayland? I tested with the OpenOrbis demo and it works
> > 
> > 
> > I'm using Hyprland and it's a wayland compositor
> 
> Just letting you know you are not alone! same issue here. But well this same error also happens on shadps4 from cli. For now I am using flatpak!

What game was this? Maybe I can try and reproduce

ryand56 committed:
shadps4: 0.13.0 -> 0.15.0

https://github.com/shadps4-emu/shadPS4/releases/tag/v.0.15.0
Diff: https://github.com/shadps4-emu/shadPS4/compare/v.0.13.0...v.0.15.0

ryand56 committed:
shadps4: 0.13.0 -> 0.15.0

https://github.com/shadps4-emu/shadPS4/releases/tag/v.0.15.0
Diff: https://github.com/shadps4-emu/shadPS4/compare/v.0.13.0...v.0.15.0

SuperSandro2000 commented on pkgs/by-name/sh/shadps4/emulator.nix:
```diff
@@ -0,0 +1,160 @@
+    };
+  };
+
+  meta.platforms = lib.intersectLists lib.platforms.linux lib.platforms.x86_64;
```
I was mainly looking for license, maintainer and homepage.

ryand56 replied:
They exist here: https://github.com/NixOS/nixpkgs/pull/474696/changes#diff-194d2f80e087120239d4842ab08c186cbf450b807c5fedafa43f53b3cc0a6f56R116

marcin-serwin commented on pkgs/by-name/sh/shadps4-qt/package.nix:
```diff
@@ -0,0 +1,114 @@
+    inherit (shadps4.meta)
+      platforms
+      license
+      maintainers
```
You should ask if they are interested in maintaining this. cc @liberodark

liberodark replied:
Hi,

Yes, I'd be happy to continue working on it, even though I'm less active on this package at the moment.
If no one else is taking care of it, I'll try to maintain it.

Best Regards

EdwinLeeford commented:
> It looks like it couldn't find the audio libraries so there was no audio. This seems to fix it for me but I'm not sure this is the best way to do this.
> 
> ```nix
> qtWrapperArgs = [
>   "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [libpulseaudio pipewire]}"
> ];
> ```

Audio is also broken for me in the "raw" shadps4 package.
```
[Lib.AudioOut] <Error> (MOMainThread) sdl_audio_out.cpp:275 OpenDevice: Failed to create SDL audio stream: Audio subsystem is not initialized
[Lib.AudioOut] <Error> (MOMainThread) sdl_audio_out.cpp:67 SDLPortBackend: Failed to initialize SDL audio backend
```

I used steam-run as a workaround but a similar fix could possibly solve this.

Sirius902 commented:
> > It looks like it couldn't find the audio libraries so there was no audio. This seems to fix it for me but I'm not sure this is the best way to do this.
> > ```nix
> > qtWrapperArgs = [
> >   "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [libpulseaudio pipewire]}"
> > ];
> > ```
> 
> Audio is also broken for me in the "raw" shadps4 package.
> 
> ```
> [Lib.AudioOut] <Error> (MOMainThread) sdl_audio_out.cpp:275 OpenDevice: Failed to create SDL audio stream: Audio subsystem is not initialized
> [Lib.AudioOut] <Error> (MOMainThread) sdl_audio_out.cpp:67 SDLPortBackend: Failed to initialize SDL audio backend
> ```
> 
> I used steam-run as a workaround but a similar fix could possibly solve this.

I noticed this as well and ended up fixing it; I believe the `LD_LIBRARY_PATH` should be on `shadps4` instead. I rebased my changes onto the new version of this PR. Also `zenity` is required or else `shadps4` will crash when it tries to open a `zenity` dialog. Here's the new patch if you want to include this. @ryand56

```diff
diff --git a/pkgs/by-name/sh/shadps4-qt/package.nix b/pkgs/by-name/sh/shadps4-qt/package.nix
index c606dfeb27d9..34c38ef5a052 100644
--- a/pkgs/by-name/sh/shadps4-qt/package.nix
+++ b/pkgs/by-name/sh/shadps4-qt/package.nix
@@ -6,9 +6,6 @@
 
   qt6,
 
-  libpulseaudio,
-  pipewire,
-
   shadps4,
   symlinkJoin,
 }:
@@ -103,15 +100,6 @@ stdenv.mkDerivation (finalAttrs: {
     runHook postFixup
   '';
 
-  qtWrapperArgs = [
-    "--prefix LD_LIBRARY_PATH : ${
-      lib.makeLibraryPath [
-        libpulseaudio
-        pipewire
-      ]
-    }"
-  ];
-
   meta = {
     inherit (shadps4.meta)
       platforms
diff --git a/pkgs/by-name/sh/shadps4/package.nix b/pkgs/by-name/sh/shadps4/package.nix
index f6287381cceb..464fed1b0986 100644
--- a/pkgs/by-name/sh/shadps4/package.nix
+++ b/pkgs/by-name/sh/shadps4/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  makeWrapper,
 
   nixosTests,
   alsa-lib,
@@ -42,6 +43,7 @@
   vulkan-memory-allocator,
   xbyak,
   xxHash,
+  zenity,
   zlib-ng,
   zydis,
   nix-update-script,
@@ -120,6 +122,7 @@ stdenv.mkDerivation (finalAttrs: {
   nativeBuildInputs = [
     cmake
     pkg-config
+    makeWrapper
   ];
 
   cmakeFlags = [
@@ -138,6 +141,15 @@ stdenv.mkDerivation (finalAttrs: {
     install -Dm644 -t $out/share/applications $src/dist/net.shadps4.shadPS4.desktop
     install -Dm644 -t $out/share/metainfo $src/dist/net.shadps4.shadPS4.metainfo.xml
 
+    wrapProgram $out/bin/shadps4 \
+      --prefix LD_LIBRARY_PATH : ${
+        lib.makeLibraryPath [
+          libpulseaudio
+          pipewire
+        ]
+      } \
+      --prefix PATH : ${lib.makeBinPath [ zenity ]}
+
     runHook postInstall
   '';
 

```

greydot commented:
I nuked the shadPS4 and shadPS4-qt configs, tried running this PR from scratch, and it segfaults.

On a related note, launching shadps4-qt from the terminal complains about missing pipewire symbols.

```
~ shadps4-qt 
qt.multimedia.symbolsresolver: Couldn't load pipewire-0.3 library
qt.multimedia.symbolsresolver: Couldn't resolve pipewire-0.3 symbols
```

Segfault stacktraces:
```
           PID: 230473 (shadps4-qt)
           UID: 1000 (void)
           GID: 100 (users)
        Signal: 11 (SEGV)
     Timestamp: Wed 2026-04-15 20:40:14 UTC (5s ago)
  Command Line: /run/current-system/sw/bin/shadps4-qt
    Executable: /nix/store/wy74md5avq408g11bkhyjcq366iwidd7-shadps4-qt-224/bin/shadps4-qt
 Control Group: /user.slice/user-1000.slice/user@1000.service/app.slice/app-net.shadps4.shadps4\x2dqtlauncher@606c772d211b443ab3c50fc35467842c.service
          Unit: user@1000.service
     User Unit: app-net.shadps4.shadps4\x2dqtlauncher@606c772d211b443ab3c50fc35467842c.service
         Slice: user-1000.slice
     Owner UID: 1000 (void)
       Boot ID: ade7ed4412d34bd89feb0b3aa2eb5a79
    Machine ID: 9dc36b05283e4a0d9de703e07c627dd7
      Hostname: apostle
       Storage: /var/lib/systemd/coredump/core.shadps4-qt.1000.ade7ed4412d34bd89feb0b3aa2eb5a79.230473.1776285614000000.zst (present)
  Size on Disk: 5.8M
       Message: Process 230473 (shadps4-qt) of user 1000 dumped core.
                
                Module /nix/store/wy74md5avq408g11bkhyjcq366iwidd7-shadps4-qt-224/bin/shadps4-qt without build-id.
                Module libjasper.so.7 without build-id.
                Module libLerc.so.4 without build-id.
                Module libtiff.so.6 without build-id.
                Module libmng.so.2 without build-id.
                Module libdav1d.so.7 without build-id.
                Module libyuv.so without build-id.
                Module libavif.so.16 without build-id.
                Module libbrotlienc.so.1 without build-id.
                Module libjxl_cms.so.0.11 without build-id.
                Module libjxl_threads.so.0.11 without build-id.
                Module libjxl.so.0.11 without build-id.
                Module libdeflate.so.0 without build-id.
                Module libImath-3_2.so.30 without build-id.
                Module libOpenEXRCore-3_3.so.32 without build-id.
                Module libIex-3_3.so.32 without build-id.
                Module libIlmThread-3_3.so.32 without build-id.
                Module libOpenEXR-3_3.so.32 without build-id.
                Module libsharpyuv.so.0 without build-id.
                Module libwebp.so.7 without build-id.
                Module libwebpmux.so.3 without build-id.
                Module libwebpdemux.so.2 without build-id.
                Module libatomic.so.1 without build-id.
                Module libvmaf.so.3 without build-id.
                Module libnuma.so.1 without build-id.
                Module libaom.so.3 without build-id.
                Module libde265.so without build-id.
                Module libx265.so.215 without build-id.
                Module libheif.so.1 without build-id.
                Module liblcms2.so.2 without build-id.
                Module libraw.so.23 without build-id.
                Module libjpeg.so.62 without build-id.
                Module libkvantum.so without build-id.
                Module libpciaccess.so.0 without build-id.
                Module libxml2.so.16 without build-id.
                Module libdrm_intel.so.1 without build-id.
                Module libdrm_amdgpu.so.1 without build-id.
                Module libxshmfence.so.1 without build-id.
                Module libxcb-sync.so.1 without build-id.
                Module libsensors.so.5 without build-id.
                Module libxcb-present.so.0 without build-id.
                Module libxcb-dri3.so.0 without build-id.
                Module libgbm.so.1 without build-id.
                Module libdrm.so.2 without build-id.
                Module libxcb-shm.so.0 without build-id.
                Module libxcb-xfixes.so.0 without build-id.
                Module libxcb-randr.so.0 without build-id.
                Module libX11-xcb.so.1 without build-id.
                Module liblzma.so.5 without build-id.
                Module libimobiledevice-glue-1.0.so.0 without build-id.
                Module libusbmuxd-2.0.so.7 without build-id.
                Module libltdl.so.7 without build-id.
                Module libvorbisfile.so.3 without build-id.
                Module libplist-2.0.so.4 without build-id.
                Module libimobiledevice-1.0.so.6 without build-id.
                Module libcanberra.so.0 without build-id.
                Module libXrender.so.1 without build-id.
                Module libxcb-keysyms.so.1 without build-id.
                Module libXfixes.so.3 without build-id.
                Module libxcb-res.so.0 without build-id.
                Module libattr.so.1 without build-id.
                Module libacl.so.1 without build-id.
                Module libXcursor.so.1 without build-id.
                Module libunistring.so.5 without build-id.
                Module libselinux.so.1 without build-id.
                Module libpsl.so.5 without build-id.
                Module libssh2.so.1 without build-id.
                Module libidn2.so.0 without build-id.
                Module libnghttp2.so.14 without build-id.
                Module libngtcp2.so.16 without build-id.
                Module libngtcp2_crypto_ossl.so.0 without build-id.
                Module libnghttp3.so.9 without build-id.
                Module libmp3lame.so.0 without build-id.
                Module libmpg123.so.0 without build-id.
                Module libogg.so.0 without build-id.
                Module libopus.so.0 without build-id.
                Module libvorbisenc.so.2 without build-id.
                Module libvorbis.so.0 without build-id.
                Module libFLAC.so.14 without build-id.
                Module libXdmcp.so.6 without build-id.
                Module libXau.so.6 without build-id.
                Module libduktape.so.207 without build-id.
                Module libsndfile.so.1 without build-id.
                Module libgomp.so.1 without build-id.
                Module libgraphite2.so.3 without build-id.
                Module libxcb.so.1 without build-id.
                Module libexpat.so.1 without build-id.
                Module libbz2.so.1 without build-id.
                Module libpcre2-8.so.0 without build-id.
                Module libffi.so.8 without build-id.
                Module libpxbackend-1.0.so without build-id.
                Module libbrotlicommon.so.1 without build-id.
                Module libkeyutils.so.1 without build-id.
                Module libkrb5support.so.0 without build-id.
                Module libcom_err.so.3 without build-id.
                Module libk5crypto.so.3 without build-id.
                Module libkrb5.so.3 without build-id.
                Module libpulsecommon-17.0.so without build-id.
                Module libpcre2-16.so.0 without build-id.
                Module libb2.so.1 without build-id.
                Module libdouble-conversion.so.3 without build-id.
                Module libicudata.so.76 without build-id.
                Module libicuuc.so.76 without build-id.
                Module libicui18n.so.76 without build-id.
                Module libGLdispatch.so.0 without build-id.
                Module libXext.so.6 without build-id.
                Module libfreetype.so.6 without build-id.
                Module libmd4c.so.0 without build-id.
                Module libharfbuzz.so.0 without build-id.
                Module libpng16.so.16 without build-id.
                Module libX11.so.6 without build-id.
                Module libfontconfig.so.1 without build-id.
                Module libEGL.so.1 without build-id.
                Module libvulkan.so without build-id.
                Module libproxy.so.1 without build-id.
                Module libz.so.1 without build-id.
                Module libzstd.so.1 without build-id.
                Module libbrotlidec.so.1 without build-id.
                Module libgssapi_krb5.so.2 without build-id.
                Module libpulse.so.0 without build-id.
                Module libxkbcommon.so.0 without build-id.
                Module libgcc_s.so.1 without build-id.
                Module libstdc++.so.6 without build-id.
                Module libOpenGL.so.0 without build-id.
                Module libGLX.so.0 without build-id.
                Module libfmt.so.12 without build-id.
                Stack trace of thread 230473:
                #0  0x00007fbe7d153dfd _ZNK9QLineEdit4textEv (libQt6Widgets.so.6 + 0x375dfd)
                #1  0x00005577d94da59e _ZN17GameInstallDialog4SaveEv (/nix/store/wy74md5avq408g11bkhyjcq366iwidd7-shadps4-qt-224/bin/shadps4-qt + 0x19e59e)
                #2  0x00007fbe7b201789 _Z10doActivateILb0EEvP7QObjectiPPv (libQt6Core.so.6 + 0x22f789)
                #3  0x00007fbe7d1c25b0 _ZN23QDialogButtonBoxPrivate19handleButtonClickedEv (libQt6Widgets.so.6 + 0x3e45b0)
                #4  0x00007fbe7b201789 _Z10doActivateILb0EEvP7QObjectiPPv (libQt6Core.so.6 + 0x22f789)
                #5  0x00007fbe7d0dc3ca _ZN15QAbstractButton7clickedEb (libQt6Widgets.so.6 + 0x2fe3ca)
                #6  0x00007fbe7d0dc683 _ZN22QAbstractButtonPrivate11emitClickedEv (libQt6Widgets.so.6 + 0x2fe683)
                #7  0x00007fbe7d0ddf93 _ZN22QAbstractButtonPrivate5clickEv (libQt6Widgets.so.6 + 0x2fff93)
                #8  0x00007fbe7d0de213 _ZN15QAbstractButton17mouseReleaseEventEP11QMouseEvent (libQt6Widgets.so.6 + 0x300213)
                #9  0x00007fbe7cff4178 _ZN7QWidget5eventEP6QEvent (libQt6Widgets.so.6 + 0x216178)
                #10 0x00007fbe7cf8544f _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent (libQt6Widgets.so.6 + 0x1a744f)
                #11 0x00007fbe7cf9025b _ZN12QApplication6notifyEP7QObjectP6QEvent (libQt6Widgets.so.6 + 0x1b225b)
                #12 0x00007fbe7b191d28 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent (libQt6Core.so.6 + 0x1bfd28)
                #13 0x00007fbe7cf8ea19 _ZN19QApplicationPrivate14sendMouseEventEP7QWidgetP11QMouseEventS1_S1_PS1_R8QPointerIS0_Ebb (libQt6Widgets.so.6 + 0x1b0a19)
                #14 0x00007fbe7d009bf2 _ZN13QWidgetWindow16handleMouseEventEP11QMouseEvent (libQt6Widgets.so.6 + 0x22bbf2)
                #15 0x00007fbe7d00ccc3 _ZN13QWidgetWindow5eventEP6QEvent (libQt6Widgets.so.6 + 0x22ecc3)
                #16 0x00007fbe7cf8544f _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent (libQt6Widgets.so.6 + 0x1a744f)
                #17 0x00007fbe7b191d28 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent (libQt6Core.so.6 + 0x1bfd28)
                #18 0x00007fbe7b9329db _ZN22QGuiApplicationPrivate17processMouseEventEPN29QWindowSystemInterfacePrivate10MouseEventE (libQt6Gui.so.6 + 0x2109db)
                #19 0x00007fbe7b9a1304 _ZN22QWindowSystemInterface22sendWindowSystemEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt6Gui.so.6 + 0x27f304)
                #20 0x00007fbe7bf3c124 _ZL23userEventSourceDispatchP8_GSourcePFiPvES1_ (libQt6Gui.so.6 + 0x81a124)
                #21 0x00007fbe7a5a6bab g_main_context_dispatch_unlocked (libglib-2.0.so.0 + 0x62bab)
                #22 0x00007fbe7a5aa188 g_main_context_iterate_unlocked.isra.0 (libglib-2.0.so.0 + 0x66188)
                #23 0x00007fbe7a5aaa2f g_main_context_iteration (libglib-2.0.so.0 + 0x66a2f)
                #24 0x00007fbe7b4d07cd _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt6Core.so.6 + 0x4fe7cd)
                #25 0x00007fbe7b19fd2b _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt6Core.so.6 + 0x1cdd2b)
                #26 0x00007fbe7d24deb7 _ZN7QDialog4execEv (libQt6Widgets.so.6 + 0x46feb7)
                #27 0x00005577d940f006 main (/nix/store/wy74md5avq408g11bkhyjcq366iwidd7-shadps4-qt-224/bin/shadps4-qt + 0xd3006)
                #28 0x00007fbe7aa44285 __libc_start_call_main (libc.so.6 + 0x2b285)
                #29 0x00007fbe7aa44338 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x2b338)
                #30 0x00005577d940f455 _start (/nix/store/wy74md5avq408g11bkhyjcq366iwidd7-shadps4-qt-224/bin/shadps4-qt + 0xd3455)
                
                Stack trace of thread 230478:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab3084 __syscall_cancel (libc.so.6 + 0x9a084)
                #3  0x00007fbe7ab30d36 ppoll (libc.so.6 + 0x117d36)
                #4  0x00007fbe7a5aa278 g_main_context_iterate_unlocked.isra.0 (libglib-2.0.so.0 + 0x66278)
                #5  0x00007fbe7a5aaa2f g_main_context_iteration (libglib-2.0.so.0 + 0x66a2f)
                #6  0x00007fbe7b4d07cd _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt6Core.so.6 + 0x4fe7cd)
                #7  0x00007fbe7b19fd2b _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt6Core.so.6 + 0x1cdd2b)
                #8  0x00007fbe7b2cca3a _ZN7QThread4execEv (libQt6Core.so.6 + 0x2faa3a)
                #9  0x00007fbe7a918881 _ZN22QDBusConnectionManager3runEv (libQt6DBus.so.6 + 0x45881)
                #10 0x00007fbe7b37b41f _ZN14QThreadPrivate5startEPv (libQt6Core.so.6 + 0x3a941f)
                #11 0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #12 0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230479:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab3084 __syscall_cancel (libc.so.6 + 0x9a084)
                #3  0x00007fbe7ab3078e __poll (libc.so.6 + 0x11778e)
                #4  0x00007fbe761e6ac2 _ZN15QtWaylandClient11EventThread3runEv (libQt6WaylandClient.so.6 + 0x9bac2)
                #5  0x00007fbe7b37b41f _ZN14QThreadPrivate5startEPv (libQt6Core.so.6 + 0x3a941f)
                #6  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #7  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230480:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab3084 __syscall_cancel (libc.so.6 + 0x9a084)
                #3  0x00007fbe7ab3078e __poll (libc.so.6 + 0x11778e)
                #4  0x00007fbe761e6ac2 _ZN15QtWaylandClient11EventThread3runEv (libQt6WaylandClient.so.6 + 0x9bac2)
                #5  0x00007fbe7b37b41f _ZN14QThreadPrivate5startEPv (libQt6Core.so.6 + 0x3a941f)
                #6  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #7  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230482:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab61ee pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x9d1ee)
                #4  0x00007fbe65183c49 cnd_wait (libgallium-26.0.4.so + 0x73ec49)
                #5  0x00007fbe6513761c util_queue_thread_func (libgallium-26.0.4.so + 0x6f261c)
                #6  0x00007fbe65183b78 impl_thrd_routine (libgallium-26.0.4.so + 0x73eb78)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230484:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab61ee pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x9d1ee)
                #4  0x00007fbe65183c49 cnd_wait (libgallium-26.0.4.so + 0x73ec49)
                #5  0x00007fbe6513761c util_queue_thread_func (libgallium-26.0.4.so + 0x6f261c)
                #6  0x00007fbe65183b78 impl_thrd_routine (libgallium-26.0.4.so + 0x73eb78)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230483:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab61ee pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x9d1ee)
                #4  0x00007fbe65183c49 cnd_wait (libgallium-26.0.4.so + 0x73ec49)
                #5  0x00007fbe6513761c util_queue_thread_func (libgallium-26.0.4.so + 0x6f261c)
                #6  0x00007fbe65183b78 impl_thrd_routine (libgallium-26.0.4.so + 0x73eb78)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230481:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab61ee pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x9d1ee)
                #4  0x00007fbe65183c49 cnd_wait (libgallium-26.0.4.so + 0x73ec49)
                #5  0x00007fbe6513761c util_queue_thread_func (libgallium-26.0.4.so + 0x6f261c)
                #6  0x00007fbe65183b78 impl_thrd_routine (libgallium-26.0.4.so + 0x73eb78)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230485:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab61ee pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x9d1ee)
                #4  0x00007fbe65183c49 cnd_wait (libgallium-26.0.4.so + 0x73ec49)
                #5  0x00007fbe6513761c util_queue_thread_func (libgallium-26.0.4.so + 0x6f261c)
                #6  0x00007fbe65183b78 impl_thrd_routine (libgallium-26.0.4.so + 0x73eb78)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230499:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab6400 pthread_cond_timedwait@@GLIBC_2.3.2 (libc.so.6 + 0x9d400)
                #4  0x00007fbe7b384093 _ZN14QWaitCondition4waitEP6QMutex14QDeadlineTimer (libQt6Core.so.6 + 0x3b2093)
                #5  0x00007fbe7b2d12ca _ZN17QThreadPoolThread3runEv (libQt6Core.so.6 + 0x2ff2ca)
                #6  0x00007fbe7b37b41f _ZN14QThreadPrivate5startEPv (libQt6Core.so.6 + 0x3a941f)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230486:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab61ee pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x9d1ee)
                #4  0x00007fbe65183c49 cnd_wait (libgallium-26.0.4.so + 0x73ec49)
                #5  0x00007fbe6513761c util_queue_thread_func (libgallium-26.0.4.so + 0x6f261c)
                #6  0x00007fbe65183b78 impl_thrd_routine (libgallium-26.0.4.so + 0x73eb78)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230498:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab6400 pthread_cond_timedwait@@GLIBC_2.3.2 (libc.so.6 + 0x9d400)
                #4  0x00007fbe7b384093 _ZN14QWaitCondition4waitEP6QMutex14QDeadlineTimer (libQt6Core.so.6 + 0x3b2093)
                #5  0x00007fbe7b2d12ca _ZN17QThreadPoolThread3runEv (libQt6Core.so.6 + 0x2ff2ca)
                #6  0x00007fbe7b37b41f _ZN14QThreadPrivate5startEPv (libQt6Core.so.6 + 0x3a941f)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230501:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab6400 pthread_cond_timedwait@@GLIBC_2.3.2 (libc.so.6 + 0x9d400)
                #4  0x00007fbe7b384093 _ZN14QWaitCondition4waitEP6QMutex14QDeadlineTimer (libQt6Core.so.6 + 0x3b2093)
                #5  0x00007fbe7b2d12ca _ZN17QThreadPoolThread3runEv (libQt6Core.so.6 + 0x2ff2ca)
                #6  0x00007fbe7b37b41f _ZN14QThreadPrivate5startEPv (libQt6Core.so.6 + 0x3a941f)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230500:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab6400 pthread_cond_timedwait@@GLIBC_2.3.2 (libc.so.6 + 0x9d400)
                #4  0x00007fbe7b384093 _ZN14QWaitCondition4waitEP6QMutex14QDeadlineTimer (libQt6Core.so.6 + 0x3b2093)
                #5  0x00007fbe7b2d12ca _ZN17QThreadPoolThread3runEv (libQt6Core.so.6 + 0x2ff2ca)
                #6  0x00007fbe7b37b41f _ZN14QThreadPrivate5startEPv (libQt6Core.so.6 + 0x3a941f)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                
                Stack trace of thread 230487:
                #0  0x00007fbe7aabf922 __syscall_cancel_arch (libc.so.6 + 0xa6922)
                #1  0x00007fbe7aab300c __internal_syscall_cancel (libc.so.6 + 0x9a00c)
                #2  0x00007fbe7aab378c __futex_abstimed_wait_common (libc.so.6 + 0x9a78c)
                #3  0x00007fbe7aab61ee pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x9d1ee)
                #4  0x00007fbe65183c49 cnd_wait (libgallium-26.0.4.so + 0x73ec49)
                #5  0x00007fbe6513761c util_queue_thread_func (libgallium-26.0.4.so + 0x6f261c)
                #6  0x00007fbe65183b78 impl_thrd_routine (libgallium-26.0.4.so + 0x73eb78)
                #7  0x00007fbe7aab6d53 start_thread (libc.so.6 + 0x9dd53)
                #8  0x00007fbe7ab3e63c __clone3 (libc.so.6 + 0x12563c)
                ELF object binary architecture: AMD x86-64
```

ryand56 commented:
Weird, I can't reproduce that segfault, both run fine for me on Wayland

greydot commented:
> Weird, I can't reproduce that segfault, both run fine for me on Wayland

Oh that's most curious. I use this [nixpkgs code](https://github.com/greydot/nixpkgs/tree/nixos-unstable-shadps4), the most recent `nixos-unstable` at the moment plus the two of your commits cherry-picked on top.

KDE+Wayland setup.

greydot commented:
> > Weird, I can't reproduce that segfault, both run fine for me on Wayland
> 
> Oh that's most curious. I use this [nixpkgs code](https://github.com/greydot/nixpkgs/tree/nixos-unstable-shadps4), the most recent `nixos-unstable` at the moment plus the two of your commits cherry-picked on top.
> 
> KDE+Wayland setup.

Ok I built shadps4-qt from git using their supplied shell.nix and it works just fine, both the latest master commit and `v224` tag. I'll dig this further.

greydot commented:
The bug is in `void GameInstallDialog::Save()`, namely that `versionDirectory` gets initialised with garbage:

```
(gdb) p addonsDirectory
$2 = {d = {d = 0x55555695cc60, ptr = 0x55555695cc70 u"/storage/games/PS4/DLCs", size = 23}}
(gdb) p versionDirectory
$3 = {d = {d = 0x7ffffffd2920, ptr = 0x7ffffffd18c0 u"鹿噡啕", size = 140736817273648}}
```

Seems like an upstream issue overall, I managed to reproduce it in a clean build. Fixed with the following patch:

```
diff --git a/src/qt_gui/game_install_dialog.cpp b/src/qt_gui/game_install_dialog.cpp
index 8a9b94d..919d4a5 100644
--- a/src/qt_gui/game_install_dialog.cpp
+++ b/src/qt_gui/game_install_dialog.cpp
@@ -144,7 +144,9 @@ void GameInstallDialog::Save() {
     // Check games directory.
     auto gamesDirectory = m_gamesDirectory->text();
     auto addonsDirectory = m_addonsDirectory->text();
+#ifndef HIDE_VERSION_MANAGER
     auto versionDirectory = m_versionDirectory->text();
+#endif
 
     if (gamesDirectory.isEmpty() || !QDir(gamesDirectory).exists() ||
         !QDir::isAbsolutePath(gamesDirectory)) {
@@ -167,6 +169,7 @@ void GameInstallDialog::Save() {
         }
     }
 
+#ifndef HIDE_VERSION_MANAGER
     if (versionDirectory.isEmpty() || !QDir::isAbsolutePath(versionDirectory)) {
         QMessageBox::critical(this, tr("Error"),
                               "The value for location to install emulator versions is not valid.");
@@ -181,11 +184,14 @@ void GameInstallDialog::Save() {
             return;
         }
     }
+#endif
 
     // Save the directories
     Config::addGameInstallDir(Common::FS::PathFromQString(gamesDirectory));
     Config::setAddonInstallDir(Common::FS::PathFromQString(addonsDirectory));
+#ifndef HIDE_VERSION_MANAGER
     m_gui_settings->SetValue(gui::vm_versionPath, versionDirectory);
+#endif
 
     const auto config_dir = Common::FS::GetUserPath(Common::FS::PathType::UserDir);
     Config::save(config_dir / "config.toml");
```

It's up to you to include it in this PR. I have it in my flake already and I'll submit it upstream some time later.

Cheers!

marcin-serwin commented:
Can we upstream the patch?

ryand56 commented:
I opened a PR upstream

marcin-serwin commented on pkgs/by-name/sh/shadps4-qt/package.nix:
```diff
@@ -0,0 +1,123 @@
+    qt6.wrapQtAppsHook
+  ];
+
+  buildInputs = (shadps4.buildInputs or [ ]) ++ [
```
You include a whole lot of `buildInputs` just to use `fmt` and `toml11`. Just decouple these packages completely, this will also simplify overriding. Same for `nativeBuildInputs`.
```suggestion
  buildInputs = [
    fmt
    toml11
```

marcin-serwin commented on pkgs/by-name/sh/shadps4-qt/package.nix:
```diff
@@ -0,0 +1,123 @@
+
+      pushd externals
+      git submodule update --init --recursive \
+        sdl3 \
```
Can we use `sdl3` from nixpkgs?

ryand56 replied:
I've tried with sdl3 from nixpkgs, unfortunately it doesn't build without the vendored one.

marcin-serwin replied:
What error are you getting? It builds fine for me with these changes:

```diff
diff --git a/pkgs/by-name/sh/shadps4-qt/package.nix b/pkgs/by-name/sh/shadps4-qt/package.nix
index c6a2bf60dc6b..ce96db1c621e 100644
--- a/pkgs/by-name/sh/shadps4-qt/package.nix
+++ b/pkgs/by-name/sh/shadps4-qt/package.nix
@@ -8,6 +8,7 @@
   qt6,
 
   fmt,
+  sdl3,
   toml11,
 
   shadps4,
@@ -20,7 +21,7 @@ stdenv.mkDerivation (finalAttrs: {
     owner = "shadps4-emu";
     repo = "shadps4-qtlauncher";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-D+1X51S3q8ZrjNN66w+MEO5C79VmXmlwbrd72GMh1Ag=";
+    hash = "sha256-lRZH9fokUKN/n3m/ZkTsRHwkwZZ04buvqBMXYLrqqLE=";
 
     postCheckout = ''
       cd "$out"
@@ -29,7 +30,6 @@ stdenv.mkDerivation (finalAttrs: {
       date -u -d "@$(git log -1 --pretty=%ct)" "+%Y-%m-%dT%H:%M:%SZ" > $out/SOURCE_DATE_EPOCH
 
       git -C externals submodule update --init --recursive \
-        sdl3 \
         volk \
         json
     '';
@@ -70,6 +70,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   buildInputs = [
     fmt
+    sdl3
     toml11
     qt6.qtbase
     qt6.qttools
```

ryand56 replied:
Okay, it builds with it now, my bad. I also added sdl3 and sdl3-mixer to the emulator along with patches for a couple of renamed functions.

marcin-serwin commented on pkgs/by-name/sh/shadps4-qt/package.nix:
```diff
@@ -0,0 +1,123 @@
+}:
+
+let
+  shadps4Wrapped = symlinkJoin {
```
Why are we doing `symlinkJoin` here? Can't we interpolate `versions.json` and `qt_ui.ini` directly into `qt-paths.patch`?

marcin-serwin commented on pkgs/by-name/sh/shadps4/package.nix:
```diff
@@ -49,29 +46,46 @@
-      find "$out" -name .git -print0 | xargs -0 rm -rf
+
+      pushd externals
+      git submodule update --init --recursive \
```
Some of those, e.g. `sdl3`, `sdl3-mixer`, `zydis`, `sirit`, `discord-rpc`, are available in nixpkgs. Can we use them instead of the vendored versions?

ryand56 replied:
I tried with the nixpkgs versions, they all don't work.

ryand56 replied:
zydis, sirit, and discord-rpc are all custom submodules for shadPS4 as seen in the repo.

marcin-serwin commented on pkgs/by-name/sh/shadps4/package.nix:
```diff
@@ -49,29 +46,46 @@
       date -u -d "@$(git log -1 --pretty=%ct)" "+%Y-%m-%dT%H:%M:%SZ" > $out/SOURCE_DATE_EPOCH
-      find "$out" -name .git -print0 | xargs -0 rm -rf
+
+      pushd externals
```
NIT: `git` supports `-C dir` option that could be used instead of pushd/popd.

marcin-serwin commented on pkgs/by-name/sh/shadps4/package.nix:
```diff
@@ -49,29 +46,46 @@
+        aacdec/fdk-aac
+      popd
+
+      find . -name .git -print0 | xargs -0 rm -rf
```
This is not necessary with `postCheckout`
```suggestion
```

marcin-serwin commented on pkgs/by-name/sh/shadps4/package.nix:
```diff
@@ -49,29 +46,46 @@
-    fetchSubmodules = true;
+    hash = "sha256-79mEvfL4ls7wSRg0DjxSIcI9ZskryIrRUTfeHbvx6NA=";
 
     leaveDotGit = true;
```
```suggestion
```

marcin-serwin commented on pkgs/by-name/sh/shadps4-qt/package.nix:
```diff
@@ -0,0 +1,123 @@
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-D+1X51S3q8ZrjNN66w+MEO5C79VmXmlwbrd72GMh1Ag=";
+
+    leaveDotGit = true;
```
```suggestion
```

marcin-serwin commented on pkgs/by-name/sh/shadps4-qt/package.nix:
```diff
@@ -0,0 +1,123 @@
+        json
+      popd
+
+      find . -name .git -print0 | xargs -0 rm -rf
```
```suggestion
```

marcin-serwin commented on pkgs/by-name/sh/shadps4/package.nix:
```diff
@@ -153,6 +163,15 @@ gcc14Stdenv.mkDerivation (finalAttrs: {
+          pipewire
+        ]
+      } \
+      --prefix PATH : ${lib.makeBinPath [ zenity ]}
```
The `zenity` is probably necessary due to using the vendored `sdl3`, it should work without this if the one from nixpkgs is used.

marcin-serwin commented on pkgs/by-name/sh/shadps4-qt/package.nix:
```diff
@@ -0,0 +1,123 @@
+    runHook preFixup
+
+    substituteInPlace $out/share/applications/net.shadps4.shadps4-qtlauncher.desktop \
+      --replace-fail 'Exec=shadPS4QtLauncher' 'Exec=shadps4-qt'
```
Why do we change the official binary name?

ryand56 replied:
I just changed the binary name back.

marcin-serwin commented on pkgs/by-name/sh/shadps4-qt/package.nix:
```diff
@@ -0,0 +1,123 @@
+    (replaceVars ./qt-paths.patch {
+      shadps4 = shadps4Wrapped;
+    })
+    ./version-directory.patch
```
```suggestion
    # https://github.com/shadps4-emu/shadps4-qtlauncher/pull/335
    ./version-directory.patch
```

marcin-serwin commented on pkgs/by-name/sh/shadps4-qt/package.nix:
```diff
@@ -0,0 +1,117 @@
+  shadps4,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "shadps4-qt";
```
Can we rename it to `shadps4-qtlauncher` to match upstream repo name?

ryand56 replied:
Done

marcin-serwin commented on pkgs/by-name/sh/shadps4-qt/package.nix:
```diff
@@ -0,0 +1,117 @@
+
+    install -Dm644 $src/.github/shadps4.png $out/share/icons/hicolor/512x512/apps/net.shadps4.shadPS4.png
+    install -Dm644 -t $out/share/applications $src/dist/net.shadps4.shadps4-qtlauncher.desktop
+    install -Dm644 -t $out/share/metainfo $src/dist/net.shadps4.shadps4-qtlauncher.metainfo.xml
```
The upstream install script already does a lot of that. We should avoid overriding `installPhase` and just do the necessary additional steps (i.e. `substitute`s for versions) in `postInstall`.

ryand56 replied:
Done

marcin-serwin reviewed (changes requested):
Sorry for so many rounds of review. It looks good to me otherwise so I hope this one is the last one.

marcin-serwin reviewed (approved):
Diff looks good and the launcher works though I don't have any games to test the emulator itself.

I'll give others a week to review and merge after that.
